library/src/main/java/com/google/android/exoplayer/hls/HlsChunkSource.java

```diff
@@ -718,13 +767,52 @@ private boolean shouldRerequestLiveMediaPlaylist(int nextVariantIndex) {
     return timeSinceLastMediaPlaylistLoadMs >= (mediaPlaylist.targetDurationSecs * 1000) / 2;
   }
 
-  private int getLiveStartChunkMediaSequence(int variantIndex) {
+
+  private int getLiveChunkMediaSequence(int variantIndex, long playbackPositionUs, boolean isReset) {
     // For live start playback from the third chunk from the end.
     HlsMediaPlaylist mediaPlaylist = variantPlaylists[variantIndex];
-    int chunkIndex = mediaPlaylist.segments.size() > 3 ? mediaPlaylist.segments.size() - 3 : 0;
+    int chunkIndex;
+    // playbackPositionUs 0 is a special case of seeking/reseting to start playback from
+    // live edge
+    if (playbackPositionUs == 0) {
+      chunkIndex = mediaPlaylist.segments.size() > LIVE_EDGE_CHUNK_INDEX_OFFSET ?
+              mediaPlaylist.segments.size() - LIVE_EDGE_CHUNK_INDEX_OFFSET : 0;
+    } else {
+      availableRange[selectedVariantIndex].getCurrentBoundsUs(availableRangeBoundsUs);
+      long relativePlaybackPositionUs = Math.abs(playbackPositionUs)  - availableRangeBoundsUs[0];
+      if (isReset) {
+        // for seek case we try to find the right chunk index.
+        // if seek position is less than lower bound we cap it at lower bound.
+        // if seek position is greater than upper bound, we cap it at the live edge
+        if (playbackPositionUs > availableRangeBoundsUs[1]) {
```

Comment:
The problem is Util.binarySearchFloor caps the return value to the last index. So, it is not possible for us to detect if the playback has gone past the upper bound using this function. 

We do not allow seeking beyond the Live edge offset (otherwise, it may result into re-buffering), but we should allow the function to return the chunk indexes till the last chunk (and beyond) for normal playback case. Hence the conditions for true and false cases of isRest have to be handled quire differently.